collector/supervisord.go

```diff
@@ -112,21 +107,45 @@ func (c *supervisordCollector) Update(ch chan<- prometheus.Metric) error {
 		StderrLogfile string `xmlrcp:"stderr_logfile"`
 		PID           int    `xmlrpc:"pid"`
 	}
-	if err := c.client.Call("supervisor.getAllProcessInfo", nil, &infos); err != nil {
-		return err
+
+	res, err := xmlrpc.Call(*supervisordURL, "supervisor.getAllProcessInfo")
+	if err != nil {
+		return fmt.Errorf("unable to call supervisord: %s", err)
 	}
-	for _, info := range infos {
+
+	for _, p := range res.(xmlrpc.Array) {
+		for k, v := range p.(xmlrpc.Struct) {
+			switch k {
```

Comment:
Just slightly worried here that field names might change in the future or might not always be present, and then we won't even notice, struct field values will just be subtly wrong. Could we guard against that by somehow checking that all expected fields are present? Simplest would be to have a `map[string]struct{}` as a set, then explicitly set an entry for a field when it has been seen, then in the end count that `len(myStruct) == 9`, and if not, return an error?

Reply:
Yea, I'm also not so happy about this way of mapping. The original library wrote directly to xml struct so we were sure the fields were mapped.